tests/cpp/test_combine_mul_sum.cpp

```diff
@@ -269,6 +271,344 @@ TEST_F(CombineMulSumAsMmaTest, UseMatmulScheduler) {
+
+  // Verify that we no longer set up MmaOp in matmul()
+  ASSERT_TRUE(ir_utils::getOpsOfType<MmaOp>(fusion.get()).empty());
+
+  auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+  std::vector<int64_t> A_shape(A_dim, batch_size);
+  A_shape[A_dim - 1] = K;
+  if (A_dim > 1) {
+    A_shape[A_dim - 2] = M;
+  }
+  at::Tensor t0 = at::randn(A_shape, options);
+  std::vector<int64_t> B_shape(B_dim, batch_size);
+  B_shape[B_dim - 1] = K;
+  if (B_dim > 1) {
+    B_shape[B_dim - 2] = N;
+  }
+  auto t1 = at::randn(B_shape, options);
+  if (transpose_a_alloc) {
+    t0 = t0.as_strided({M, K}, {1, M});
+  }
+  std::vector<c10::IValue> inputs{t0, t1};
+  at::Tensor tref;
+  if (bias_dim >= 0) {
+    at::Tensor bias;
+    if (bias_dim == 0) {
+      bias = at::randn({}, options);
+    } else if (bias_dim == 1) {
+      bias = at::randn({N}, options);
+    } else {
+      NVF_ERROR(false, "Invalid bias dimension given:", bias_dim);
+    }
+    inputs.emplace_back(bias);
+    tref = at::linear(t0, t1, bias);
+  } else {
+    tref = at::linear(t0, t1);
+  }
+  tref = tref.sin().to(at::kHalf);
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  auto outputs = executor_cache.runFusionWithInputs(inputs);
+
+  const FusionKernelRuntime* runtime =
+      executor_cache.getMostRecentKernelRuntime();
+  ASSERT_NE(runtime, nullptr);
+
+  if (expect_aten_eval) {
+    EXPECT_TRUE(runtime->isSegmented());
+  } else {
+    EXPECT_FALSE(runtime->isSegmented());
+  }
+
+  ScheduleHeuristic heuristic =
+      runtime->schedulerHeuristics()->heuristicsList().front()->heuristic();
+  if (expect_aten_eval) {
+    EXPECT_EQ(heuristic, ScheduleHeuristic::ExprEval);
+  } else {
+    // Ensure that the Matmul scheduler ran.
+    // Assert here since we will inspect the kernel next, which we can't
+    // do if ExprEval accepts the segment.
+    ASSERT_EQ(heuristic, ScheduleHeuristic::Matmul);
+    // Ensure there's an MmaOp.
+    EXPECT_FALSE(
+        ir_utils::getOpsOfType<MmaOp>(runtime->executors().at(0).kernel())
+            .empty());
+  }
+
+  testValidate(
+      executor_cache.fusion(), outputs, inputs, {tref}, __LINE__, __FILE__);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    ,
+    LinearNodeTranslationTest,
+    ::testing::Values(
+        // Tests without fusion enabled
+        std::make_tuple(2l, 2l, -1l, false, false, true),
+        std::make_tuple(2l, 2l, -1l, false, true, true),
+        // Enable fusion
+        std::make_tuple(2l, 2l, -1l, true, false, false),
+        // We cannot yet handle allocation domain in matmul scheduler
+        std::make_tuple(2l, 2l, -1l, true, true, true),
+        // We don't fuse 1D inputs
+        std::make_tuple(1l, 2l, -1l, true, false, true),
+        std::make_tuple(2l, 1l, -1l, true, false, true),
+        // TODO: The following currently fails but it should not be translated
+        // to LinearOp to begin with
+        // std::make_tuple(1l, 1l, -1l, true, false, false),
```

Comment:
Why does this fail? Based on our offline conversation we decided to pipe everything through ATen. I do need to remove the special cases from MatmulOp.

Reply:
The NoOp scheduler accepts the whole fusion, which looks like this:
```
Fusion IR after pre-segmenter optimization passes:
Inputs:
  T0_g[ iS0{i0} ], __half
  T1_g[ iS3{i0} ], __halfIt should be easy to fix the NoOp scheduler to avoid accepting zero-dimensional non-trivial fusions.
Outputs:
  T5_g[ ], __half

%kernel_math {
T2_l[ rS2{i0} ]
   = linear(T0_g[ iS0{i0} ],
            T1_g[ iS3{i0} ]  )
T3_l[ ]
   = __half2float(T2_l[ rS2{i0} ]);
T4_l[ ]
   = sinf(T3_l[ ]);
T5_g[ ]
   = __float2half(T4_l[ ]);
}
```
I just updated the NoOp scheduler to avoid fusions with matmul ops, which fixes this issue. I now check cases like this with and without fusion enabled.